java/server/src/org/openqa/selenium/remote/server/rest/ResultConfig.java

```diff
@@ -166,6 +168,34 @@ public Response handle(Command command) throws Exception {
     return response;
   }
 
+  private String removeCredentials(String command) {
+    if (Strings.isNullOrEmpty(command)) {
+      return command;
+    }
+
+    if (command.contains("send keys")) {
+      java.util.regex.Pattern credentialsPattern =
+        java.util.regex.Pattern
+          .compile("(.*)(send keys:).*(])", java.util.regex.Pattern.CASE_INSENSITIVE);
+      Matcher matcher = credentialsPattern.matcher(command);
+
+      if (matcher.matches()) {
+        return matcher.group(1) + matcher.group(2) + " REDACTED" + matcher.group(3);
```

Comment:
If the regex matches, are we guaranteed that both the groups will be present? What if one of the groups turns out to be null?

Reply:
Yup, the matches method only returns true when all groups are matched. In this case there would be 3 groups. It is possible that a group is empty but it wont be null.